### PR TITLE
Added support for C# 9 record type

### PR DESCRIPTION
**Pull request by DrPepperBianco**
Fixed Issue #42 

Added override for `VisitRecordDeclaration` to add suuport for C# records.

Records are added as `class <<record>>` to the PlantUML output.

Other than `class` record can have a parameter list. This creates a constructor with that parameter list as well as properties. I only added properties to the PlantUML output, but not the constructors.

I also updated the README.md accordingly.

### CODE REVIEW

**Comment by gromag** [comment]
I'm also looking forward to record addition.

**Comment by dominikb** [comment]
Just tried to use this package in VSCode with records and got the following error
image

When I built the repo with changes from the PR it worked just fine. As far as I'm concerned this PR looks good to merge 👍

**Comment by MathieuDR** [comment]
Is there a reason why this is being blocked?

**Comment by svdHero** [comment]
We need this support for C# records, too. This is crucial for all modern C# projects. @pierre3 could you merge this, please? I would be ever so grateful.

**Comment by Vivraan** [comment]
@pierre3 tagging again for visibility.

**Comment by Benoit-FM-Shift** [comment]
Hello,
This branch  has been marge with Main?

**Comment by MathieuDR** [comment]
Perhaps @DrPepperBianco could create a new nuget package with the updated code. As @pierre3 seems to be unavailble / not maining this anymore.

**Comment by Vivraan** [comment]
Pinging for attention, again.

@DrPepperBianco @pierre3

File: src/PlantUmlClassDiagramGenerator.Library/ClassDiagramGenerator.cs [hunk 1]
```diff
@@ -59,6 +59,76 @@ public override void VisitClassDeclaration(ClassDeclarationSyntax node)
...
+            var type = $"{name}{typeParam}";
+
+            types.Add(name);
+
+            // Write records as: "class <<record>>"
+            WriteLine($"class {type} {modifiers}<<record>> {{");
+
+            nestingDepth++;
+
+            foreach(var parameter in node.ParameterList.Parameters)
```
Comment (radu-v) [inline src/PlantUmlClassDiagramGenerator.Library/ClassDiagramGenerator.cs:87]: `node.ParameterList` will be null for records with no parameters (e.g. when a record only has explicit properties), so maybe a null check would be beneficial here

**Comment by MathieuDR** [comment]
Thanks @DrPepperBianco. Do you feel comfortable opening issues in your repo & perhaps setting up a new nuget package feed?

File: src/PlantUmlClassDiagramGenerator.Library/ClassDiagramGenerator.cs (hunk 1 above)
Comment (radu-v) [reply src/PlantUmlClassDiagramGenerator.Library/ClassDiagramGenerator.cs:87]: apparently `Microsoft.CodeAnalysis.CSharp` needed updating - version `4.4.0-2.final` supports `record struct`.

**Comment by pierre3** [comment]
@DrPepperBianco
Thank you for your patience.

I added test code and fixed some issues, and merged.
- Added support for "record struct".
- Fixed the issue that generics type properties are not output

**Pull request merged**